tools/shared.py

```diff
@@ -177,20 +177,19 @@ def run_process(cmd, check=True, input=None, *args, **kw):
 
   debug_text = '%sexecuted %s' % ('successfully ' if check else '', ' '.join(cmd))
 
-  if hasattr(subprocess, 'run'):
-    # Python 3.5 and above only
-    kw.setdefault('encoding', 'utf-8')
-    result = subprocess.run(cmd, check=check, input=input, *args, **kw)
-  else:
-    # Python 2 compatibility: Introduce Python 3 subprocess.run-like behavior
-    if input is not None:
-      kw['stdin'] = subprocess.PIPE
-    proc = Popen(cmd, *args, **kw)
-    stdout, stderr = proc.communicate(input)
-    result = Py2CompletedProcess(cmd, proc.returncode, stdout, stderr)
-    if check:
-      result.check_returncode()
+  if hasattr(subprocess, "run"):
+    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
+    logger.debug(debug_text)
```

Comment:
Not sure why this line was removed in the original PR, but would it make sense to keep it removed? It looks like it doesn't have anything to do with the bugs in question.

Reply:
Not sure, good question. Can leave for @sbc100 when he can take a look.